unity-renderer/Assets/Scripts/MainScripts/DCL/Components/UI/UIReferencesContainer.cs

```diff
@@ -25,6 +26,19 @@ public class UIReferencesContainer : MonoBehaviour, IPointerDownHandler
 
         bool VERBOSE = false;
 
+        private bool hasParentRectTransform = false;
+        private RectTransform parentRectTransform;
+
+        public RectTransform GetParentRectTransform()
+        {
+            if (!hasParentRectTransform)
+            {
+                parentRectTransform = GetComponentInParent<RectTransform>();
+                hasParentRectTransform = true;
+            }
+            return parentRectTransform;
+        }
```

Comment:
you can replace it with lazy property in the way

```
private RectTransform parentRectTransformLazy;
public RectTransform ParentRectTransform =>  parentRectTransformLazy ??=  GetComponentInParent<RectTransform>();

```

Reply:
I prefer not to opt for this lazy approach because public RectTransform GetParentRectTransform() is frequently called in the layout refresh, and checking a bool saves some time over checking null.
Also, using the null-coalescing operator may generate (in this specific case) false positives and give a different result than checking == null (which would slow down the layout refresh).